packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -1019,7 +1007,7 @@ export function transformCjsImport(
     importNames.forEach(({ importedName, localName }) => {
       if (importedName === '*') {
         lines.push(
-          `const ${localName} = (${interopHelper.toString()})(${cjsModuleName})`,
+          `const ${localName} = ${cjsModuleName} && ${cjsModuleName}.__esModule ? ${cjsModuleName} : { ...(typeof ${cjsModuleName} === 'object' && !Array.isArray(${cjsModuleName}) ? ${cjsModuleName} : {}), default: ${cjsModuleName} }`,
```

Comment:
Same here, I don't know if we should use `m` in both cases for the param name so it is easier to follow (instead of `d` above)
```suggestion
          `const ${localName} = (m => m?.__esModule ? m : { ...(typeof m === 'object' && !Array.isArray(m) ? m : {}), default: m })(${cjsModuleName})`,
```